main.cc

```diff
@@ -544,6 +544,85 @@ static locator::host_id initialize_local_info(sharded<db::system_keyspace>& sys_
     return linfo.host_id;
 }
 
+static bool raft_topology_change_enabled(const db::config& cfg) {
+    return cfg.consistent_cluster_management() &&
+        cfg.check_experimental(db::experimental_features_t::feature::CONSISTENT_TOPOLOGY_CHANGES);
+}
+
+static void initialize_system_keyspace(sharded<db::system_keyspace>& sys_ks,
```

Comment:
@xemul should we make scylla_main() a class? though it's not a good fit, the constructor yields like crazy.

Reply:
> @xemul should we make scylla_main() a class? though it's not a good fit, the constructor yields like crazy.

I think it would be great to have a single class for wiring scylla components together, so that we can reuse it in `scylla_main` and boost tests.